### PR TITLE
We now have a main menu!

### PR DESCRIPTION
**Pull request by Schtekt**
The game scene now displays a menu where the user may start the game or exit with mouseclicks.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/MainMenuHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    Scene* scene = sm->CreateScene("MainMenuScene");
+
+    component::GUI2DComponent* guic = nullptr;
+
+    Texture* startTex = al->LoadTexture2D(L"../Vendor/Resources/Textures/2DGUI/Start.png");
+    Texture* optionsTex = al->LoadTexture2D(L"../Vendor/Resources/Textures/2DGUI/Options.png");
+    Texture* exitTex = al->LoadTexture2D(L"../Vendor/Resources/Textures/2DGUI/Exit.png");
+
+    // Player (Need a camera)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/MainMenuHandler.cpp:41]: Love it ❤️

File: StortSpelProjekt/Game/src/main.cpp [hunk 2]
```diff
@@ -609,6 +610,52 @@ Scene* ShopScene(SceneManager* sm)
     return scene;
 }
 
+void logSchtuff(const std::string& name)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/main.cpp]: What is this?

File: StortSpelProjekt/Game/src/main.cpp [hunk 3]
```diff
@@ -609,6 +610,52 @@ Scene* ShopScene(SceneManager* sm)
...
+{
+    Log::Print("%s\n", name.c_str());
+}
+
+Scene* MainMenuScene(SceneManager* sm)
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/main.cpp]: Why is this scene created in two places?

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
Make the mouse be released as default while in menu

File: StortSpelProjekt/Game/src/main.cpp (hunk 2 above)
Comment (Schtekt) [reply StortSpelProjekt/Game/src/main.cpp]: Something that should not exist, thank you for spotting it 👍

File: StortSpelProjekt/Game/src/main.cpp (hunk 3 above)
Comment (Schtekt) [reply StortSpelProjekt/Game/src/main.cpp]: it was a copy pasta from main instead of cut and pasta. The function is removed from main.

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
Menu buttons can be clicked while in game, even though not visible

**Review by LeoWikstrom — DISMISSED** [review]
It is approved

**Review by jocke1995 — DISMISSED** [review]
so good, yes so perfect! yes yes yes!

File: StortSpelProjekt/Game/src/Gamefiles/Components/TeleportComponent.cpp [hunk 4]
```diff
@@ -38,6 +38,7 @@ void component::TeleportComponent::OnCollision(Collision* collisionEvent)
 		else
 		{
 			Player::GetInstance().IsInShop(false);
+			EventBus::GetInstance().Publish(&RoundStart());
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/TeleportComponent.cpp:41]: Jacob told me a few weeks back that its usually bad practise to publish events inside events, did you think this structure through before you did this?

File: StortSpelProjekt/Game/src/main.cpp [hunk 5]
```diff
@@ -52,11 +53,10 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR, int nCmdShow)
     Scene* demoScene = GameScene(sceneManager);
     Scene* shopScene = ShopScene(sceneManager);
     Scene* gameOverScene = GameOverScene(sceneManager);
+    Scene* mainMenuScene = MainMenuHandler::GetInstance().CreateScene(sceneManager);
 
-    //Scene* shopScene = ShopScene(sceneManager);
-    sceneManager->SetScene(demoScene);
+    sceneManager->SetScenes(mainMenuScene);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/main.cpp]: Its not called SetScenes any more. its called SetScene. This gave me compilation error.

File: StortSpelProjekt/Engine/src/ECS/SceneManager.cpp [hunk 6]
```diff
@@ -116,7 +116,7 @@ void SceneManager::ChangeScene()
 {
 	if (m_ChangeSceneNextFrame)
 	{
-		if (m_pActiveScene->GetName() == "ShopScene" || m_pActiveScene->GetName() == "GameScene")
+		if (m_ActiveScenes[0]->GetName() != "gameOverScene")
```
Comment (jocke1995) [inline StortSpelProjekt/Engine/src/ECS/SceneManager.cpp]: this should be m_pActiveScene, also gave me a compilation error

**Review by jocke1995 — DISMISSED** [review]
Almost! Just a few compilation errors

File: StortSpelProjekt/Game/src/main.cpp (hunk 5 above)
Comment (Schtekt) [reply StortSpelProjekt/Game/src/main.cpp]: Fixed

File: StortSpelProjekt/Engine/src/ECS/SceneManager.cpp (hunk 6 above)
Comment (Schtekt) [reply StortSpelProjekt/Engine/src/ECS/SceneManager.cpp]: Fixed

File: StortSpelProjekt/Game/src/Gamefiles/Components/TeleportComponent.cpp (hunk 4 above)
Comment (Schtekt) [reply StortSpelProjekt/Game/src/Gamefiles/Components/TeleportComponent.cpp:41]: The only other option I see to this is to add an update to the teleportcomponent that checks if a boolean is true or not to call the roundstart event. For that reason I don't see another solution to this problem. I would like to know why this is bad practice though?

**Review by LeoWikstrom — APPROVED** [review]

**Review by OtakuFreak13 — APPROVED** [review]

**Pull request merged**